frontend/src/actions/plans.ts

```diff
@@ -0,0 +1,143 @@
+"use server";
+
+import { revalidatePath } from "next/cache";
+
+import { auth, fetchToAdonis } from "@/lib/auth";
+
+interface CreatePlanResponseType {
+  success: boolean;
+  message: string;
+  schedule: {
+    name: string;
+    userId: number;
+    createdAt: string;
+    updatedAt: string;
+    id: number;
+  };
+}
+
+interface PlanResponseType {
+  name: string;
+  userId: number;
+  id: number;
+  createdAt: string;
+  updatedAt: string;
+  courses: Array<{
+    id: string;
+    name: string;
+    department: string;
+    lecturer: string;
+    type: string;
+    ects: number;
+    semester: number;
+    groups: Array<{
+      id: number;
+      name: string;
+      day: string;
+      time: string;
+      room: string;
+    }>;
+  }>;
+  registrations: Array<{
+    id: string;
+    name: string;
+  }>;
+}
+
+interface DeletePlanResponseType {
+  success: boolean;
+  message: string;
+}
+
+export const createNewPlan = async ({
+  name,
+  courses,
+  registrations,
+  groups,
+}: {
+  name: string;
+  courses: Array<{ id: string }>;
+  registrations: Array<{ id: string }>;
+  groups: Array<{ id: number }>;
+}) => {
+  const isLogged = await auth({});
+  if (!isLogged) {
+    return false;
+  }
+
+  const data = await fetchToAdonis<CreatePlanResponseType>({
+    url: "/user/schedules",
+    method: "POST",
+    body: JSON.stringify({ name, courses, registrations, groups }),
+  });
+  if (!data) {
+    return false;
+  }
+  return data;
+};
+
+export const updatePlan = async ({
+  id,
+  name,
+  courses,
+  registrations,
+  groups,
+}: {
+  id: number;
+  name: string;
+  courses: Array<{ id: string }>;
+  registrations: Array<{ id: string }>;
+  groups: Array<{ id: number }>;
+}) => {
+  const isLogged = await auth({});
+  if (!isLogged) {
+    return false;
+  }
+
+  const data = await fetchToAdonis<CreatePlanResponseType>({
+    url: `/user/schedules/${id}`,
+    method: "PATCH",
+    body: JSON.stringify({ name, courses, registrations, groups }),
+  });
+  if (!data) {
+    return false;
+  }
```

Comment:
dlaczego zwracasz false, to jakaś dziwna praktyka, rzuć błędem po prostu jak status code jest niepoprawny